lib/src/stream_queue.dart

```diff
@@ -228,17 +228,19 @@ abstract class StreamQueue<T> {
   ///
   /// Until the transaction finishes, this queue won't emit any events.
   ///
+  /// See also [withTransaction] and [cancelable].
+  ///
   /// ```dart
   /// /// Consumes all empty lines from the beginning of [lines].
-  /// Future consumeEmptyLines(StreamQueue<String> lines) {
+  /// Future consumeEmptyLines(StreamQueue<String> lines) async {
   ///   while (await lines.hasNext) {
   ///     var transaction = lines.startTransaction();
   ///     var queue = transaction.newQueue();
   ///     if ((await queue.next).isNotEmpty) {
   ///       transaction.reject();
   ///       return;
   ///     } else {
-  ///       transaction.accept(queue);
+  ///       transaction.commit(queue);
```

Comment:
Maybe we should have a `peek` method on `StreamQueue`. That would make this much easier.